src/server/game/Auction/AsyncAuctionOperation.cpp

```diff
@@ -0,0 +1,502 @@
+/*
+ * This file is part of the AzerothCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by the
+ * Free Software Foundation; either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AsyncAuctionOperation.h"
+#include "AuctionHouseMgr.h"
+#include "AuctionHousePackets.h"
+#include "Creature.h"
+#include "DatabaseEnv.h"
+#include "GameTime.h"
+#include "ObjectAccessor.h"
+#include "Player.h"
+#include "ScriptMgr.h"
+
+Player* AsyncAuctionOperation::GetPlayer() const
+{
+    return ObjectAccessor::FindPlayer(_playerGuid);
+}
+
+void SellItemTask::Execute()
+{
+    auto player{ GetPlayer() };
+    if (!player)
+        return;
+
+    auto session{ player->GetSession() };
```

Comment:
@Winfidonarleyan It looks like async tasks are still thread unsafe.
If I understood correctly this task can be executed in parallel with a map update for example.
And if that is true, then you can't use player/session objects in several threads without synchronisation mechanisms (mutexes/locks/semaphores).
In your shoes I would try to copy all required player/session related data into a new object(s) during enqueueing/scheduling a task. And use this new object inside of these tasks instead of real player/session objects.